src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java

```diff
@@ -595,12 +600,22 @@ private boolean tryCell(int cell, int point, int delta)
         }
     }
 
-    private static int roundKey(int p, int roundSeconds)
+    private static int roundKey(int point, int roundSeconds)
     {
-        int d = p % roundSeconds;
+        int d = point % roundSeconds;
         if (d > 0)
-            return p + (roundSeconds - d);
+        {
+            point += roundSeconds - d;
+            if (point == Cell.MAX_DELETION_TIME)
```

Comment:
This should be >

Reply:
Fixed. The new testcase "testThatPointIsNotMissedBecauseOfRoundingToNoDeletionTime" added to cover this branch